pkg/plugins/profile/pd_profile_handler.go

```diff
@@ -91,8 +93,10 @@ func (h *PdProfileHandler) WithName(name string) *PdProfileHandler {
 // previously executed cycles along with their results.
 func (h *PdProfileHandler) Pick(ctx context.Context, cycleState *types.CycleState, request *types.LLMRequest, profiles map[string]*framework.SchedulerProfile,
 	profileResults map[string]*types.ProfileRunResult) map[string]*framework.SchedulerProfile {
+	startTime := time.Now()
 	if _, executed := profileResults[h.decodeProfile]; !executed {
 		// if decode profile was not executed yet, first let the scheduler run the decode profile
+		defer func() { metrics.RecordDecodeSelectionDuration(time.Since(startTime)) }()
```

Comment:
This is incorrect, it will call the function when the profile is returned to the scheduler not after it ran.